
camera_frame: 'camera_link'
object_frame: 'aruco_box'
robot_frame: 'base_link'
odom_frame: 'odom'
map_frame: 'map'

target_id: 8
aruco_size: 0.048
desired_corner_locations: [[381., 435.], 
                            [243., 435.], 
                            [240., 301.], 
                            [384., 302.]]

z_desired: 0.18

focal_length_pixel: 530.466941

camera_matrix: [[530.466941, 0, 205], 
                [0, 530.466941, 154], 
                [0, 0, 1]]

distortion_coeffs: [0.0, 0.0, 0.0, 0.0, 0.0]

target_detection_topic: '/corners'
ibvs_activate_topic: '/ibvs_activate'
pose_control_activate_topic: '/pose_control_activate'
orientation_control_activate_topic: '/orientation_control_activate'
pose_controller_topic: '/goal_point'
orientation_controller_topic: '/orientation_goals'
camera_topic: '/camera/image_raw'
unlock_topic: '/unlock'
control_rate: 50
navigator_rate: 50
commands_topic: '/cmd_vel'
odometry_topic: '/map'
# sim_odom_topic: '/odom'
map_topic: '/map'

uncertainty_localisation:
  wl_topic: 'wl'
  wr_topic: 'wr'
  k_l: 0.25
  k_r: 0.25
  odom_rate: 60


map_path: '/home/puzzlebot/catkin_ws/src/T3003B_IntelligentRobotics/LidarWorkspace/slam/maps/real_arena_landmarks.yaml'
map_localisation:
  scan_topic: '/scan'
  odom_topic: '/map'
  lidar_resolution: 1147
  lidar_offset: -90 # Lidar offset is -90 for real lidar and 90 for simulated in gazebo.
  num_ransac_iterations: 75
  ransac_d_threshold: 0.06
  min_ransac_inliers: 60
  search_range: 60
  inertial_frame_name: 'map'
  landmark_distance_threshold: 0.25

kalman_localisation:
  wl_topic: 'wl'
  wr_topic: 'wr'
  k_l: 0.05
  k_r: 0.05
  odom_rate: 60

pose_controller:
  odom_topic: '/map'

  # Control parameters
  k1: .5
  k2: 7
  beta: 0.8
  lambda_: 1.0
  v_max: 0.15

  # Tolerances
  r_tolerance: 0.05
  
ibvs_controller:
  # Proportional controller
  kp: 0.44
  # Tolerances
  e_tolerance: 30

orientation_controller:
  odom_topic: '/map'
  kp: 0.01
  ki: 0.001
  e_tolerance: 0.01
